admin/grants.sql

```diff
@@ -498,3 +498,188 @@ GRANT ROLE SYNAPSE_DATA_WAREHOUSE_ANALYST TO SYNAPSE_DATA_WAREHOUSE_ADMIN;
 -- SYNAPSE_DATA_WAREHOUSE_DEV domain roles
 GRANT ROLE SYNAPSE_DATA_WAREHOUSE_DEV_ADMIN TO SYSADMIN;
 GRANT ROLE SYNAPSE_DATA_WAREHOUSE_DEV_ANALYST TO SYNAPSE_DATA_WAREHOUSE_DEV_ADMIN;
+
+---- RBAC reconfiguration of data warehouse ----
+-- The following grants transfer ownership of current and future 
+-- data warehouse objects from SYSADMIN to each namespace's respective
+-- `*_ALL_ADMIN` database role.
+
+---- SYNAPSE_DATA_WAREHOUSE ----
+
+-- SYNAPSE
+GRANT OWNERSHIP ON SCHEMA SYNAPSE_DATA_WAREHOUSE.SYNAPSE
+	TO DATABASE ROLE SYNAPSE_DATA_WAREHOUSE.SYNAPSE_ALL_ADMIN
+	COPY CURRENT GRANTS;
+GRANT OWNERSHIP ON TABLES
+	IN SCHEMA SYNAPSE_DATA_WAREHOUSE.SYNAPSE
+	TO DATABASE ROLE SYNAPSE_DATA_WAREHOUSE.SYNAPSE_ALL_ADMIN
+	COPY CURRENT GRANTS;
+GRANT OWNERSHIP ON FUTURE TABLES
+	IN SCHEMA SYNAPSE_DATA_WAREHOUSE.SYNAPSE
+	TO DATABASE ROLE SYNAPSE_DATA_WAREHOUSE.SYNAPSE_ALL_ADMIN;
+GRANT OWNERSHIP ON DYNAMIC TABLES
+	IN SCHEMA SYNAPSE_DATA_WAREHOUSE.SYNAPSE
+	TO DATABASE ROLE SYNAPSE_DATA_WAREHOUSE.SYNAPSE_ALL_ADMIN
+	COPY CURRENT GRANTS;
+GRANT OWNERSHIP ON FUTURE DYNAMIC TABLES
+	IN SCHEMA SYNAPSE_DATA_WAREHOUSE.SYNAPSE
+	TO DATABASE ROLE SYNAPSE_DATA_WAREHOUSE.SYNAPSE_ALL_ADMIN;
+GRANT OWNERSHIP ON STAGES
+	IN SCHEMA SYNAPSE_DATA_WAREHOUSE.SYNAPSE
+	TO DATABASE ROLE SYNAPSE_DATA_WAREHOUSE.SYNAPSE_ALL_ADMIN
+	COPY CURRENT GRANTS;
+GRANT OWNERSHIP ON FUTURE STAGES
+	IN SCHEMA SYNAPSE_DATA_WAREHOUSE.SYNAPSE
+	TO DATABASE ROLE SYNAPSE_DATA_WAREHOUSE.SYNAPSE_ALL_ADMIN;
+GRANT OWNERSHIP ON VIEWS
+	IN SCHEMA SYNAPSE_DATA_WAREHOUSE.SYNAPSE
+	TO DATABASE ROLE SYNAPSE_DATA_WAREHOUSE.SYNAPSE_ALL_ADMIN
+	COPY CURRENT GRANTS;
+GRANT OWNERSHIP ON FUTURE VIEWS
+	IN SCHEMA SYNAPSE_DATA_WAREHOUSE.SYNAPSE
+	TO DATABASE ROLE SYNAPSE_DATA_WAREHOUSE.SYNAPSE_ALL_ADMIN;
+GRANT OWNERSHIP ON TASKS
+	IN SCHEMA SYNAPSE_DATA_WAREHOUSE.SYNAPSE
+	TO DATABASE ROLE SYNAPSE_DATA_WAREHOUSE.SYNAPSE_ALL_ADMIN
+	COPY CURRENT GRANTS;
+GRANT OWNERSHIP ON FUTURE TASKS
```

Comment:
What about future streams in case we make one in the `synapse` schema for whatever reason?

Reply:
See my comment below.